### PR TITLE
Editor: Initial implementation of TabbedPanel element

### PR DESCRIPTION
**Pull request by sdinesh86**
As per the discussion in PR #17443 will be implementing a few things for the editor. this is a the first of many PR's.

This PR adds a tabbed panel in the UI.
I have sort of just moved the logic from the Sidebar.js to a new UI element and added few more methods.

Next I will be replacing the current tabbed panels with this implementation.

### CODE REVIEW

**Comment by mrdoob** [comment]
Any chance you could add a screenshot of how does this look like?

**Comment by sdinesh86** [comment]
So it looks the same its just that the code is different i will push the changes on sidebar update the PR

**Comment by mrdoob** [comment]
Oh I see.

> Next I will be replacing the current tabbed panels with this implementation.

Can you make these changes as part of this PR too?

**Comment by sdinesh86** [comment]
done you can check it

**Comment by mrdoob** [comment]
Looking good!

File: editor/js/libs/ui.js [hunk 1]
```diff
@@ -995,3 +1011,122 @@ UI.Button.prototype.setLabel = function ( value ) {
...
+
+	}
+
+	this.selected = id;
+
+	return this;
+
+}
+
+UI.TabbedPanel.prototype.addPanel = function ( id, label, items ) {
```
Comment (mrdoob) [inline editor/js/libs/ui.js]: I think `addTab` would be a better name here.

Comment (sdinesh86) [reply editor/js/libs/ui.js]: I initiallt thought of that name, but now i dont remember why i changed it :(

**Comment by mrdoob** [comment]
Thanks got changing the name. I think you forgot to update all the calls.

**Comment by sdinesh86** [comment]
OOPS

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**